src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs

```diff
@@ -318,7 +318,10 @@ public void Dispose()
             }
             if (_intermediateFolder != null)
             {
-                // to-do: add check for build status. if failed, not overwrite buildinfo
+                if (_currentBuildInfo.Versions.Any(v => v.Status == BuildStatus.Failed))
```

Comment:
i used a temporary walk around: always fully rebuild if last build failed

---
In reply to: [76365473](https://github.com/dotnet/docfx/pull/638#discussion_r76365473) [](ancestors = 76365473,76353748)